quickwit-search/src/fetch_docs.rs

```diff
@@ -150,9 +150,11 @@ async fn fetch_docs_in_split(
     index_storage: Arc<dyn Storage>,
     split: &SplitIdAndFooterOffsets,
 ) -> anyhow::Result<Vec<(GlobalDocAddress, String)>> {
-    let index_reader = get_searcher_for_split(global_doc_addrs.len(), index_storage, split).await?;
+    let num_concurrent_requests = 10;
+    let index_reader = get_searcher_for_split_without_cache(1, index_storage, split).await?;
+    let searcher = Arc::new(index_reader.searcher());
     let doc_futures = global_doc_addrs.into_iter().map(|global_doc_addr| {
```

Comment:
Nothing in this function contract tells me whether the global_doc_addrs
Maybe the caller sorts them I am not sure.

Should we sort them in this function maybe?
Or document that it should be sorted and add an assert that.

Reply:
good point, we should sort it here